### PR TITLE
Core: better error message when analyzer created without tokenizer or…

### PR DESCRIPTION
**Pull request by jeff303**
This should address #15492.  I'm not sure if I put the test case in the correct place, so I can move that if needed.  Also, I wasn't entirely sure how to exercise the other code path (AnalysisRegistry:338) so any tips on that would be appreciated.

### CODE REVIEW

**Comment by karmi** [comment]
Hi @jeff303, we have found your signature in our records, but it seems like you have signed with a different e-mail than the one used in yout Git [commit](https://github.com/elastic/elasticsearch/pull/18455.patch). Can you please add both of these e-mails into your Github profile (they can be hidden), so we can match your e-mails to your Github profile?

**Comment by jeff303** [comment]
@karmi , apologies, I forgot to update user.email for this particular commit.  I have done so, and force-pushed.

File: core/src/test/java/org/elasticsearch/action/admin/indices/create/CreateIndexIT.java [hunk 1]
```diff
@@ -277,4 +278,19 @@ public void testRestartIndexCreationAfterFullClusterRestart() throws Exception {
...
+    public void testNoTypeOrTokenizerErrorMessage() throws Exception {
+        Settings.Builder stBldr = Settings.builder();
+        stBldr.putArray("analysis.analyzer.test_analyzer.filter", new String[] {"lowercase", "stop", "shingle"});
+        stBldr.putArray("analysis.analyzer.test_analyzer.char_filter", new String[] {"html_strip"}).build();
+
+        CreateIndexRequestBuilder b = prepareCreate("test", 1, stBldr);
+        try {
+            b.get();
+            fail("IllegalArgumentException should have been thrown");
+        } catch (Exception e) {
```
Comment (s1monw) [inline core/src/test/java/org/elasticsearch/action/admin/indices/create/CreateIndexIT.java]: can you use

```Java
assertEquals(expectThrows(IllegalArgumentException.class, () -> b.get()).getMessage(), 
 "analyzer [test_analyzer] must specify either an analyzer type, or a tokenizer");
```

**Comment by s1monw** [comment]
this looks great, I left one comment on the test! 

thanks for improving this!

File: core/src/test/java/org/elasticsearch/action/admin/indices/create/CreateIndexIT.java [hunk 2]
```diff
@@ -277,4 +278,19 @@ public void testRestartIndexCreationAfterFullClusterRestart() throws Exception {
         internalCluster().fullRestart();
         ensureGreen("test");
     }
+
+    public void testNoTypeOrTokenizerErrorMessage() throws Exception {
```
Comment (s1monw) [inline core/src/test/java/org/elasticsearch/action/admin/indices/create/CreateIndexIT.java]: I think we can test this without creating an index - maybe a simple unittest in `AnalysisServiceTests` would do? To trigger the problem I think all you need to do is to build an `AnalysisService` like this:

```Java
        Version version = VersionUtils.randomVersion(random());
        Settings settings = Settings
            .builder()
            .put(IndexMetaData.SETTING_VERSION_CREATED, version)
            .put(Environment.PATH_HOME_SETTING.getKey(), createTempDir().toString())
            .putArray("analysis.analyzer.test_analyzer.filter", new String[] {"lowercase", "stop", "shingle"});
            .putArray("analysis.analyzer.test_analyzer.char_filter", new String[] {"html_strip"})
            .build();
        IndexSettings idxSettings = IndexSettingsModule.newIndexSettings("index", settings);
        AnalysisService analysisService = new AnalysisRegistry(null, new Environment(settings)).build(idxSettings);
```

**Comment by karmi** [comment]
@jeff303, great, the CLA check is now passing correctly!

**Comment by s1monw** [comment]
looks awesome! I am running tests and will pull this in afterwards

**Comment by s1monw** [comment]
merged thanks!

**Pull request merged**